doc/source/dev/contributor/using_blas_lapack.rst

```diff
@@ -0,0 +1,213 @@
+.. _using-blas-lapack:
+
+Using BLAS and LAPACK from within SciPy
+=======================================
+
+1. Cython
+2. Fortran
+3. C and C++
+
+
+Concepts, design and common issues
+----------------------------------
+
+By design, we allow using any BLAS/LAPACK library that the user or packager
+has installed on their system, and we try hard to detect or provide build
+options to handle any commonly used build configuration of known BLAS and
+LAPACK libraries in the wild.
+
+When using a BLAS or LAPACK function from within SciPy, there are a number of
+things we have to deal with:
+
+1. ABI issues. BLAS and LAPACK have a Fortran ABI - and there are two different
+   ABIs in use (gfortran, and f2c (or g77)). MKL and Accelerate use the f2c
+   ABI, while OpenBLAS, ATLAS and Netlib BLAS *usually* use the gfortran ABI.
+   However, conda-forge uses a Netlib BLAS interface with f2c ABI. Since we
+   cannot know for sure which ABI we need at build time, we choose sensible
+   defaults but also offer an explicit build option to override that default
+   (``-Duse-g77-abi``).
+2. 32-bit vs. 64-bit integers. Most libraries use 32-bit integers - this is
+   called the LP64 interface. Sometimes a library is built with 64-bit integers
+   (ILP64). It's also possible that both 32-bit and 64-bit symbols are present
+   in the same library (this is currently the case for MKL and Accelerate). To
+   accommodate ILP64, we need to use different function prototypes, changing
+   the integer parameters from ``int`` to ``npy_int64``.
+3. Symbol names. In general, the plain BLAS/LAPACK routine names are not enough
+   to know what symbol names are present in an installed library. There may be
+   a common symbol prefix and/or suffix. Examples: 
+
+   - ``scipy-openblas`` always uses a ``scipy_`` symbol prefix.
+   - The ILP64 OpenBLAS builds use a ``64_`` symbol suffix (typically;it's
+     also possible that there is no suffix).
+   - Accelerate uses a ``$NEWLAPACK`` suffix for LP64 symbols, and ``$NEWLAPACK_ILP64`` for ILP64 symbols.
```

Comment:
Interesting, is the `$` explicit/raw, or meant to be substituted like a variable? That's a weird one!

Reply:
It's an actual raw `$` as part of the symbol name.